src/libxtp/jobtopology.cc

```diff
@@ -349,5 +349,46 @@ void JobTopology::WriteToHdf5(std::string filename) const {
   }
 }
 
+void JobTopology::ReadFromHdf5(std::string filename) {
+  CheckpointFile cpf(filename, CheckpointAccessLevel::READ);
+  CheckpointReader a = cpf.getReader();
+  Index id = -1;
+  a(id, "jobid");
+  if (id != _job.getId()) {
+    throw std::runtime_error(
+        "Jobid from checkpoint file does not agree with jobid" +
+        std::to_string(id) + ":" + std::to_string(_job.getId()));
+  }
+  _regions.clear();
```

Comment:
Out of curiosity, why do you call this `clear` here?

Reply:
otherwise it could in principle append the regions from the file, so I delete all the old regions first.